homeassistant/components/myuplink/update.py

```diff
@@ -0,0 +1,78 @@
+"""Update entity for myUplink."""
+from typing import cast
+
+from homeassistant.components.update import (
+    UpdateDeviceClass,
+    UpdateEntity,
+    UpdateEntityDescription,
+    UpdateEntityFeature,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import MyUplinkDataCoordinator
+from .const import DOMAIN
+from .entity import MyUplinkEntity
+
+UPDATE_DESCRIPTION = UpdateEntityDescription(
+    key="update",
+    device_class=UpdateDeviceClass.FIRMWARE,
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up update entity."""
+    entities: list[UpdateEntity] = []
+    coordinator: MyUplinkDataCoordinator = hass.data[DOMAIN][config_entry.entry_id]
+
+    # Setup update entities
+    for device in coordinator.data.devices:
+        entities.append(
+            MyUplinkDeviceUpdate(
+                coordinator=coordinator,
+                device_id=device,
+                entity_description=UPDATE_DESCRIPTION,
+                unique_id_suffix="upd",
+            )
+        )
+
+    async_add_entities(entities)
+
+
+class MyUplinkDeviceUpdate(MyUplinkEntity, UpdateEntity):
+    """Representation of a myUplink device update entity."""
+
+    _attr_supported_features = UpdateEntityFeature(0)
+    entity_description: UpdateEntityDescription
+
+    def __init__(
+        self,
+        coordinator: MyUplinkDataCoordinator,
+        device_id: str,
+        entity_description: UpdateEntityDescription,
+        unique_id_suffix: str,
+    ) -> None:
+        """Initialize the update entity."""
+        super().__init__(
+            coordinator=coordinator,
+            device_id=device_id,
+            unique_id_suffix=unique_id_suffix,
+        )
+
+        self.entity_description = entity_description
+        self._attr_name = "Update"
```

Comment:
What is this update entity updating? If it's the firmware, please remove this line and add the device class FIRMWARE. (This can be done outside of the constructor)

Reply:
Firmware device class is in the entity description. We can just remove this line, I think.
```suggestion
```

Reply:
Oh haven't spotted that, good catch

Reply:
Removing...